streamlit_drawable_canvas/frontend/src/lib/polypath.ts

```diff
@@ -107,7 +104,9 @@ class PolypathTool extends FabricTool {
       selectable: false,
       evented: false,
     })
-    canvas.add(this.currentPath)
+    if (this.currentPath.width !== 0 && this.currentPath.height !== 0) {
+      canvas.add(this.currentPath)
+    }
```

Comment:
My solution to prevent adding zero-sized polypath. Similar to the one applied in `line.ts` but with slightly different logic.